curated_transformers/models/with_strided_spans.py

```diff
@@ -56,6 +66,14 @@ def with_strided_spans_forward(
 
     Y_layer, backprop_layer = model.layers[0](spans, is_train=is_train)
 
+    is_trf_output = isinstance(Y_layer, TransformerModelOutput)
```

Comment:
Could you explain this conditional block? In what case it the layer output not transformer output?

Reply:
At the moment, some of the tests for the `with_strided_spans` model simply pass in `Ragged` inputs. I thought it would be cleaner to support them instead of wrapping them in a `TransformerModelOutput` (and it makes the model usable outside of transformer contexts).

Reply:
I think we should initially just make `with_strided_spans` work with `TransformerModelOutput`s. I think it's better to make the tests mock some `MakeModelOutputs` than to have union types and condition blocks to handle them this early on. It makes the types harder to read, the implementations more difficult to follow, and I think it is unlikely that these layers would be used outside `curated-transformers`.